builtin/k8s/platform.go

```diff
@@ -367,6 +368,23 @@ func (p *Platform) Deploy(
 
 	timeout := 10 * time.Minute
 
+	cancelMu := &sync.Mutex{}
+	pollingComplete := false
+	go func() {
```

Comment:
A goroutine in the plugin like this is likely unsafe. After the RPC calls end, we will terminate the plugin at some point, and there is no guarantee this goroutine cleanly exits. It is likely that that process is slow enough that this _happens_ to work, but we can't guarantee it going forward.